pyvista/core/grid.py

```diff
@@ -243,7 +243,7 @@ def meshgrid(self) -> list:  # numpydoc ignore=RT01
             List of numpy arrays representing the points of this mesh.
 
         """
-        return np.meshgrid(self.x, self.y, self.z, indexing='ij')
+        return np.meshgrid(self.x, self.y, self.z, indexing='ij')  # type: ignore
```

Comment:
This isn't a problem only for the release.  It exists in the main branch too.  I'm not sure why it is only showing up in the typing here though.  I saw it in #6275 as well.

Reply:
The return type is wrong for numpy 2.0.  We need to come up with a decision on how we will type going forward.

Reply:
Maybe a naive suggestion, but you could `list(np.meshgrid(...))` or (probably better / more future-compatible) `tuple(np.meshgrid(...))`, no?

Reply:
The return type is the same the same as `np.meshgrid`.  In v 1.26, the return type of `np.meshgrid` is `builtins.list[numpy.ndarray[Any, numpy.dtype[Any]]]`.  In v 2.0, the return type is `builtins.tuple[numpy.ndarray[Any, numpy.dtype[Any]], ...]`.

My comment is that we shoudl decide whether we type on numpy version 1.26 or version 2.x.  If we ignore like this, implicitly we are saying that we are going to type numpy version 1.26.  We need to decide this.  I would vote for numpy 2.x.  But this needs to be decided on a more holistic level with the CI testing.

Also our pre-commit mypy testing using different numpy pinning than our CI testing, e.g. for pytest.  This is creating these sorts of problems.

Reply:
Will mypy be happy with typing it with both list and tuple?

Reply:
Yeah I see that the return type changes for 1.26 vs 2.0. I'm suggesting that if you change the code to:
```
        return tuple(np.meshgrid(self.x, self.y, self.z, indexing='ij'))
```
then it doesn't matter if you type on 1.26 or 2.0, the return type of this `meshgrid` method will always be `-> tuple[np.ndarray]`. But maybe `mypy` wouldn't like this, or you prefer your return type to mirror NumPy's 🤷

Reply:
I was preferring not to let the type-tail wag the code-dog here, given that this method is a thin wrapper around `np.meshgrid`. But both are valid options.  My main comment is that ignoring the type seems like the wrong solution long term, but may be viable for this release.

Reply:
> This isn't a problem only for the release. It exists in the main branch too. I'm not sure why it is only showing up in the typing here though. I saw it in #6275 as well.

This may be the same issue as #5611. `mypy` can be very finicky with imports. If not done correctly `mypy` will treat missing imports as having type `Any`, see https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports. Since `pre-commit` runs `mypy` in an isolated environment and only checks modified files, it may be that `mypy` is unable to properly import from any unmodified files, causing `mypy` to miss imports and treat objects as `Any`. Then, when `mypy` runs again later with _different_ modified files, the imports are successful and an error is raised.

This is a huge problem in my opinion. We should seriously consider dropping `mypy` from pre-commit, or keeping it but using a customized hook to make sure it does what we want. The main thing that's needed I think is to make `mypy` has full project scope for the imports, not just one or two modified files. One option to consider is creating a local pre-commit hook (see https://stackoverflow.com/a/67796237)  to run `mypy` in the local environment, e.g. by calling `mypy pyvista` in the shell instead of using the `pre-commit` environment.

Reply:
I thought about this, and I think the best solution for this particular use case is to type the return for both versions, i.e. `list` and `tuple`.  Since we are just providing a thin wrapper around `np.meshgrid` it makes sense to type based on all the versions we support.